### PR TITLE
feat(Designer): Introduce experimentation to enable A/B testing of features

### PR DESCRIPTION
**Pull request by takyyon**
## Requirement Checklist

* [x] The commit message follows our guidelines
* [ ] Tests for the changes have been added (for bug fixes or features)
* [ ] Docs have been added or updated (for bug fixes or features)

## Type of Change

* [ ] Bug fix
* [x] Feature
* [ ] Other

## Impact of Change

* [ ] **This is a breaking change.**

### CODE REVIEW

File: libs/logic-apps-shared/src/designer-client-services/lib/experimentation.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+let service: IExperimentationService;
+
+export const InitExperimentationServiceService = (experimentationService: IExperimentationService): void => {
+  service = experimentationService;
+};
+
+export const ExperimentationService = (): IExperimentationService => {
+  if (!service) {
```
Comment (ccastrotrejo) [inline libs/logic-apps-shared/src/designer-client-services/lib/experimentation.ts:16]: Just to be extra sure about this. The designer doesn't crash or throws this error if the service is not needed and not initilialized, right? for example in vscode extension

**Review by ccastrotrejo — APPROVED** [review]

File: libs/logic-apps-shared/src/designer-client-services/lib/experimentation.ts (hunk 1 above)
Comment (takyyon) [reply libs/logic-apps-shared/src/designer-client-services/lib/experimentation.ts:16]: It shouldn't. The designer will only crash in scenarios where the experimentation service is being used but not initialized. Each client using the designer needs to initialize the service. On the side I'll also be checking to see how we can always default to a certain value value here

Comment (takyyon) [reply libs/logic-apps-shared/src/designer-client-services/lib/experimentation.ts:16]: Actually, I went ahead and added it explicitly for vscode as well. We need a better way to default certain service if the value remains the same for all default experiences.

Comment (takyyon) [reply libs/logic-apps-shared/src/designer-client-services/lib/experimentation.ts:16]: On a second thought, I went ahead and always defaulted to a false implementation of ExperimentationService so incase a certain client using the designer does not want to use the A/B test feature, they won't get it unless they explicitly add in the experimentation service

Comment (ccastrotrejo) [reply libs/logic-apps-shared/src/designer-client-services/lib/experimentation.ts:16]: Thanks Krrish!

**Pull request merged**